src/jest.js

```diff
@@ -168,8 +168,9 @@ function runCLI(argv, packageRoot, onComplete) {
     var testPaths = argv.onlyChanged ?
       _promiseOnlyChangedTestPaths(testRunner, config) :
       _promisePatternMatchingTestPaths(argv, testRunner);
+    var testReporter = new (require(config.testReporter))();
     return testPaths.then(function (testPaths) {
-      return testRunner.runTests(testPaths);
+      return testRunner.runTests(testPaths, testReporter);
```

Comment:
I don't see where this is being used?

Additionally, I don't think you need to pass this in here since the `TestRunner` already has access to the `config` object -- so assuming it needs it, the TestRunner should be able to get access to the reporter internally itself